src/nbl/asset/utils/CSPIRVIntrospector.cpp

```diff
@@ -13,472 +13,528 @@ namespace nbl::asset
         {
-            auto checkedDescsCntIt = checkedDescsCnt;
-            auto introspectionIt = introspections;
-            for (auto shader : shaders)
+            if (prev.stageFlags)
             {
-                auto& introBindings = (*introspectionIt)->descriptorSetBindings[_set];
-                if (*checkedDescsCntIt!=introBindings.size())
-                {
-                    const auto& introBinding = introBindings[*checkedDescsCntIt].binding;
-                    if (introBinding==binding.binding)
-                    {
-                        stageFlags |= shader->getStage();
-                        refIndex = (*checkedDescsCntIt)++;
-                        refIntro = *introspectionIt;
-                        refShader = shader;
-                    }
-                }
-                introspectionIt++;
-                checkedDescsCntIt++;
+                prev.size = current.offset - prev.offset;
+                tmp.push_back(prev);
             }
+            prev = current;
         }
+    }
 
-        auto& introBnd = refIntro->descriptorSetBindings[_set][refIndex];
-        binding.type = resType2descType(introBnd.type);
-        binding.stageFlags = static_cast<ICPUShader::E_SHADER_STAGE>(stageFlags);
-        binding.count = introBnd.descriptorCount;
-        if (introBnd.descCountIsSpecConstant)
-        {
-            auto& specInfo = refShader->getSpecializationInfo();
-            auto val = specInfo.getSpecializationByteValue(binding.count);
-            assert(val.second == 4ull);
-            memcpy(&binding.count, val.first, 4ull);
-        }
+    if (prev.stageFlags)
+    {
+        prev.size = current.offset - prev.offset;
+        tmp.push_back(prev);
+    }
+
+    if(tmp.size() == 0)
+        return nullptr;
 
-        bindings.push_back(binding);
+    std::ostringstream debug;
+    for (auto it = tmp.begin(); it != tmp.end(); it++)
+        debug << "Stage flags: " << std::bitset<32>(it->stageFlags) << " offset: " << it->offset << " size: " << it->size << std::endl;
+    std::cout << debug.str();
+
+    return core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<SPushConstantRange>>(tmp);
+}
+NBL_API2 core::smart_refctd_ptr<ICPUDescriptorSetLayout> CSPIRVIntrospector::CPipelineIntrospectionData::createApproximateDescriptorSetLayoutFromIntrospection(const uint32_t setID)
+{
+    std::vector<ICPUDescriptorSetLayout::SBinding> outBindings;
+    outBindings.reserve(m_descriptorSetBindings[setID].size());
+
+    for (const auto& binding : m_descriptorSetBindings[setID])
+    {
+        auto& outBinding = outBindings.emplace_back();
+
+        outBinding.binding = binding.binding;
+        outBinding.count = binding.count;
+        outBinding.type = binding.type;
+        outBinding.stageFlags = binding.stageMask;
+        outBinding.createFlags = ICPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE;
+        // TODO: outBinding.samplers
```

Comment:
you cannot use immutable samplers, because HLSL doesn't give you any info about what such a sampler should be.

Reply:
remove the TODO and correct the comment